### PR TITLE
topology: add delay to setting hover states

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
Addresses usability issue trying to create a new connection.

Resolves https://jira.coreos.com/browse/ODC-2210

### CODE REVIEW

**Comment by christianvogt** [comment]
@jeff-phillips-18 So the idea is to just add a hover out delay for the connector? Because we still need a configurable delay for hover in.

**Comment by jeff-phillips-18** [comment]
@christianvogt The delay is for both in and out, and it is configurable. Didn't think we needed it to be configurable separately for the two as it acts a bit strange when you do that (two nodes highlighted briefly).

File: frontend/packages/topology/src/utils/useHover.ts [hunk 1]
```diff
@@ -25,21 +26,27 @@ const useHover = <T extends Element>(
...
+          const delayedStateChange = (newState: boolean) => {
+            clearTimeout(unsetHandle.current);
+            clearTimeout(delayHandle);
+
+            if (timeoutDelay != null) {
               delayHandle = window.setTimeout(() => {
-                delayHandle = clearTimeout(unsetHandle.current);
-                setHover(true);
-              }, delay);
+                clearTimeout(unsetHandle.current);
```
Comment (christianvogt) [inline frontend/packages/topology/src/utils/useHover.ts]: Trying to recall why this is inside the timeout and not outside. However by you clearing the timeout above, we don't need this here.

File: frontend/packages/topology/src/utils/useHover.ts [hunk 2]
```diff
@@ -25,21 +26,27 @@ const useHover = <T extends Element>(
...
+                setHover(newState);
+              }, timeoutDelay);
             } else {
               clearTimeout(unsetHandle.current);
```
Comment (christianvogt) [inline frontend/packages/topology/src/utils/useHover.ts]: This line is redundant with line 30.

File: frontend/packages/topology/src/utils/useHover.ts [hunk 3]
```diff
@@ -49,18 +56,19 @@ const useHover = <T extends Element>(
             node.removeEventListener('mouseenter', onMouseEnter);
             node.removeEventListener('mouseleave', onMouseLeave);
             clearTimeout(delayHandle);
+            delayHandle = null;
```
Comment (christianvogt) [inline frontend/packages/topology/src/utils/useHover.ts]: This isn't needed.

File: frontend/packages/topology/src/utils/useHover.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ const useHover = <T extends Element>(
 ): [boolean, (node: T) => (() => void) | undefined] => {
   const [hover, setHover] = React.useState<boolean>(false);
   const unmountRef = React.useRef(false);
+  const timeoutDelay = delay !== undefined ? delay : 200;
```
Comment (christianvogt) [inline frontend/packages/topology/src/utils/useHover.ts]: Use a default value in the param
```
delay? = 200,
```

I'd like to see both a separate `delayIn` and `delayOut` but we can add as needed in future.

File: frontend/packages/topology/src/utils/useHover.ts (hunk 1 above)
Comment (jeff-phillips-18) [reply frontend/packages/topology/src/utils/useHover.ts]: it is needed if the timer is created during the hover delay, not needed below when there is no delay.

File: frontend/packages/topology/src/utils/useHover.ts (hunk 4 above)
Comment (jeff-phillips-18) [reply frontend/packages/topology/src/utils/useHover.ts]: I've added them

**Pull request merged**